src/main/java/jpass/crypt/io/CryptInputStream.java

```diff
@@ -112,6 +113,7 @@ public CryptInputStream(InputStream parent, byte[] key, byte[] iv) {
      */
     public CryptInputStream(InputStream parent, byte[] key) throws IOException {
         this._parent = parent;
+        parent.read(new byte[36]);//uuid salt
```

Comment:
What do you think about moving the salt related reading/writing logic to a specific input/output stream, e.g. SaltInputStream/SaltOutputStream?